### PR TITLE
Add Docker for building images

### PR DESCRIPTION
**Pull request by gnadaban**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -67,7 +67,16 @@ RUN apt-get update && apt-get install -y --no-install-recommends \
...
-&& setcap cap_ipc_lock= /usr/bin/vault
+&& setcap cap_ipc_lock= /usr/bin/vault \
+# Install Docker CLI
+&& curl -fsSL https://download.docker.com/linux/ubuntu/gpg | gpg --dearmor -o /usr/share/keyrings/docker-archive-keyring.gpg \
+&& echo \
+  "deb [arch=amd64 signed-by=/usr/share/keyrings/docker-archive-keyring.gpg] https://download.docker.com/linux/ubuntu \
+  $(lsb_release -cs) stable" | tee /etc/apt/sources.list.d/docker.list > /dev/null \
+&& apt-get update \
+&& apt-get install -y --no-install-recommends \
+    docker-ce=5:20.10.\* \
```
Comment (blueskyjunkie) [inline Dockerfile:78]: Use kaniko https://github.com/GoogleContainerTools/kaniko for unprivileged docker container builds.

Comment (gnadaban) [reply Dockerfile:78]: The Azure DevOps Docker@2 tasks depend on Docker directly unfortunately.
Writing a task in Azure DevOps script that uses Kaniko will take more time, without much added value.
Are you sure we want to do that?

Comment (gnadaban) [reply Dockerfile:78]: Eh, let me take a look.

Comment (gnadaban) [reply Dockerfile:78]: Apparently to run the Kaniko build job we would still need to have the Docker CLI in the ADO image.
With that, the Azure DevOps Docker task is not versatile enough to provide file inputs to containers, so we cannot use Kaniko with Azure DevOps.

The example that describes why we need files is [here](https://blog.csanchez.org/2020/09/30/building-docker-images-with-kaniko-pushing-to-azure-container-registry-acr/).

**Pull request merged**